File: minimal.websharper.web/Startup.cs
using asplib.Model.Db;
using iie;
using Microsoft.AspNetCore;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using System;
using WebSharper.AspNetCore;

namespace minimal.websharper.web
{
    public class Startup
    {
        public IConfigurationRoot Configuration { get; }
        public IWebHostEnvironment Environment { get; }
        public IHttpContextAccessor HttpContext { get; }

        public Startup(IWebHostEnvironment env)
        {
            Environment = env;
            var dom = new ConfigurationBuilder()
                .SetBasePath(env.ContentRootPath)
                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
                .AddEnvironmentVariables()
                .Build();
            Configuration = dom;

            ASP_DBEntities.ConnectionString = Configuration["ASP_DBEntities"];  // globally

            // Inject static references into the Remoting component
            Remoting.Configuration = Configuration;
            Remoting.Environment = Environment;
        }

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddDistributedMemoryCache()
                .AddSession(options =>
                {
                    options.IdleTimeout = TimeSpan.FromMinutes(30);
                    options.Cookie.HttpOnly = true;
                })
                .AddLogging()
                // From WebSharper:
                .AddSitelet(Site.Main)
                .AddAuthentication("WebSharper")
                .AddCookie("WebSharper", options => { });
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            app.UseDeveloperExceptionPage()
                .UseSession()
                .UseMiddleware<IIEMiddleware>()
                // From WebSharper:
                .UseAuthentication()
                .UseStaticFiles()
                .UseWebSharper()
                .Run(context =>
                {
                    context.Response.StatusCode = 404;
                    return context.Response.WriteAsync("Page not found");
                });
        }

        public static void Main(string[] args)
        {
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .CaptureStartupErrors(true)
                .UseSetting("detailedErrors", "true")
                .Build()
                .Run();
        }
    }
}

File: src/mandelbrot.renderer/Startup.fs
namespace mandelbrot.renderer

open Microsoft.AspNetCore
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Logging
open Microsoft.Extensions.Hosting
open WebSharper.AspNetCore
open asplib.Remoting
open asplib.Services
open System

type Startup(configuration: IConfiguration) =

    member this.ConfigureServices(services: IServiceCollection) =
        services.AddSingleton(configuration)
            .AddImageCache()    // by Image.Key
            .AddMemoryCache()   // by Image.Gid
            .AddImageService()  // Singleton for static access
            .AddLogging()
            .AddSitelet(Site.ImageBinary)
            .AddAuthentication("WebSharper")
            .AddCookie("WebSharper", fun options -> ())
        |> ignore

    member this.OnShutDown(disposable: IDisposable) =
        disposable.Dispose()

    member this.Configure(app: IApplicationBuilder,
                            env: IWebHostEnvironment,
                            applifetime: IHostApplicationLifetime,
                            disposable: ImageService) =
        if env.IsDevelopment() then
            WebSharper.Web.Remoting.DisableCsrfProtection() |> ignore
            app.UseDeveloperExceptionPage() |> ignore
            // Thanks to context.Environment.["WebSharper.AspNetCore.Services"] only needed for tests:
            StaticServiceProvider.SetProvider(app.ApplicationServices);

        applifetime.ApplicationStopping.Register(fun _ -> this.OnShutDown(disposable)) |> ignore

        app.UseAuthentication()
            .UseImageSocket(Site.RendererWebSocket)
            .UseRequestQuerySession()
            .UseWebSharper()
            .Run(fun context ->
                context.Response.StatusCode <- 404
                context.Response.WriteAsync("Page not found"))

module Program =
    let BuildWebHost args =
        WebHost
            .CreateDefaultBuilder(args)
            .ConfigureLogging(fun logging ->
                logging.AddConsole()
                        .AddDebug() |> ignore
            )
            .UseStartup<Startup>()
            .Build()

    [<EntryPoint>]
    let main args =
        BuildWebHost(args).Run()
        0